guides/intro.md

```diff
@@ -0,0 +1,615 @@
+> another, it only takes RTP packets from a track in the first peer's PeerConnection and feeds them to another track in the second peer's PeerConnection.
+> For instance, the role of `Forwarder` in the examples above would be to forward media in such way:
+>
+> ```mermaid
+> flowchart LR
+>   subgraph Forwarder
+>     track11 -.-> track12
+>     track22 -.-> track21
+>   end
+>   PC1[PeerConnection 1] --> track11
+>   PC2[PeerConnection 2] --> track22
+>   track12 --> PC2
+>   track21 --> PC1
+> ```
+>
+> This might be a bit counterintuitive, as in reality both of the tracks `track11` and `track12` still carry the same media stream.
+
+A similar process would happen for all of the joining/leaving peers. If you want to check an actual working example, check out the
+[Nexus](https://github.com/elixir-webrtc/apps/tree/master/nexus) - our Elixir, WebRTC-based videoconferencing demo app.
+ 
+#### 2. Codecs
+
+When connecting two peers, you also have to make sure that all of them use the same video and audio codec, as the codec negotiation happens
+completely separately between independent PeerConnections.
+
+You can tell what codec is carried in an RTP packet by inspecting its payload type (`packet.payload_type`, a non-negative integer field) and match it with one
+of the codecs listed in this track's transceiver's `codecs` field (you have to find the `transceiver` by iterating over `PeerConnection.get_transceivers` as shown
+previously in this tutorial).
+
+In a real scenario, you'd have to receive the RTP packet from the PeerConnection, inspect its payload type, find the codec associated with that payload type, find the payload type
+associated with that codec on the other PeerConnection, and use it to overwrite the original payload type in the packet.
+
+Unfortunately, at the moment the `PeerConnection.send_rtp` API forces you to use the topmost negotiated codec, so there's no way to handle RTP streams with changing codecs.
+The only real solution is to force `PeerConnection` to negotiate only one codec.
+
+```elixir
+codec = %ExWebRTC.RTPCodecParameters{
+    payload_type: 96,
+    mime_type: "video/VP8",
+    clock_rate: 90_000
+}
+{:ok, pc} = PeerConnection.start_link(video_codecs: [codec])
+```
+
+This is not ideal as the remote PeerConnection might not support this particular codec. This tutorial will be appropriately updated once the `PeerConnection` API allows
+for more in this regard.
+
+#### 3. Types of vidoe frames
+
+When speaking about video codecs, we should also mention the idea of different types of frames.
+
+We are interested in these types (although there can be more, depending on the codec):
+* I-frames (/intra-frames/keyframes) - these are complete, independent images and do not require other frames to be decoded
+* P-frames (predicted frames/delta-frames) - these only hold changes in the image from the previous frame.
```

Comment:
Idea: maybe we could link https://github.com/leandromoreira/digital_video_introduction somewhere?